src/cdk-experimental/testing/protractor/protractor-element.ts

```diff
@@ -39,8 +73,31 @@ export class ProtractorElement implements TestElement {
         .perform();
   }
 
-  async sendKeys(keys: string): Promise<void> {
-    return this.element.sendKeys(keys);
+  async sendKeys(...keys: (string | TestKey)[]): Promise<void>;
+  async sendKeys(modifiers: ModifierKeys, ...keys: (string | TestKey)[]): Promise<void>;
+  async sendKeys(...modifiersAndKeys: any[]): Promise<void> {
+    const first = modifiersAndKeys[0];
+    let modifiers: ModifierKeys;
+    let rest: (string | TestKey)[];
+    if (typeof first !== 'string' && typeof first !== 'number') {
+      modifiers = first;
+      rest = modifiersAndKeys.slice(1);
+    } else {
+      modifiers = {};
+      rest = modifiersAndKeys;
+    }
+
+    const modifierKeys = [
+        modifiers.control ? Key.CONTROL : '',
+        modifiers.alt ? Key.ALT : '',
+        modifiers.shift ? Key.SHIFT : '',
+        modifiers.meta ? Key.META : ''
+    ].filter(k => k);
```

Comment:
Logic-wise this looks good, but I personally feel like it would be more clean like that:

```ts
const modifierKeys = [];
if (modifiers.control) {
  modifierKeys.push(Key.CONTROL);
}
if (modifiers.alt) {
  modifierKeys.push(Key.ALT);
}
...
```

Surely that's way more verbose, but it could be in a separate function (and also avoids the unnecessary iteration)

Reply:
Sigh, I hate doing repeated if-blocks, but I guess the filter isn't the most readable...

Reply:
Yeah totally valid point :smile: This was just my personal opinion so you can decide, but thanks for changing!